specs/waku/waku-1.md

```diff
@@ -303,7 +302,15 @@ Each node SHOULD broadcast its rate limits to its peers using the rate limits pa
 
 Each node SHOULD respect rate limits advertised by its peers. The number of packets SHOULD be throttled in order not to exceed peer's rate limits. If the limit gets exceeded, the connection MAY be dropped by the peer.
 
-##### Message Confirmations Field
+##### light Node Field
+
+Light Node represents a boolean option. When true, the peer SHOULD NOT forward Envelopes through the Messages packet (`0x01`).
+
+##### Confirmations Enabled Field
+
+Confirmations enabled represents a boolean option. When true, the peer SHOULD send [message confirmations](#batch-acknowlege-and-message-response).
```

Comment:
Perhaps same as above, perhaps not, as I don't know how confirmations work and in which direction this confirmations enabled field works.

Reply:
I checked in the `status-go` code and the only usage I can see is:

https://github.com/status-im/status-go/blob/2d13ccf5ec3db7e48d7a96a7954be57edb96f12f/waku/v1/peer.go#L535-L547

```go
	// mark envelopes only if they were successfully sent
	for _, e := range bundle {
		p.Mark(e)
		event := common.EnvelopeEvent{
			Event: common.EventEnvelopeSent,
			Hash:  e.Hash(),
			Peer:  p.EnodeID(),
		}
		if p.confirmationsEnabled {
			event.Batch = batchHash
		}
		p.host.SendEnvelopeEvent(event)
	}
```

Which seems to check if the peer's `status-option.ConfirmationsEnabled` is true. If true add the `batchHash` to the `EnvelopeEvent`. This pushes the `EnvelopeEvent` on to the event queue. The `protocol\transport\waku` event handler checks if the `event.Batch` is equal to `event.Hash`, if it is `m.handler.EnvelopeSent()` is triggered, this seems to be the trigger for sending the message confirmation.

Which seems to suggest that if a peer has a its `confirmations-enabled` set to true, the host should send a confirmation.

@cammellos sorry to bug you, is this correct?

Reply:
mmmh, this has puzzled me for a while.
 
I understood it was the other way aronud:

If a host has confirmation enabled, it will send confirmations to any peer (regardless on whether confirmations are enabled on the peer), I was looking at https://github.com/status-im/status-go/blob/2d13ccf5ec3db7e48d7a96a7954be57edb96f12f/waku/v1/peer.go#L263 

But looking at the code that you posted now makes me doubt that.

What I can see, is that in `Whisper` confirmations were enabled unless explicitly disabled:

https://github.com/status-im/status-go/blob/2d13ccf5ec3db7e48d7a96a7954be57edb96f12f/whisper/whisper.go#L104

And the comment is "do not reply with confirmations", so the intent is clear.

We then changed this to be explicitly confirmed in waku, and I noticed in the scope of the PR for images that it was never set in the config in the nodes on the fleets (I asked jakubs to enabled it), still messages were actually being confirmed, not sure exactly how/why.

I checked the client (status-react) and does not explicitly set confirmations enabled for waku (whisper they were on, as we did not set disableConfirmations).

I think there's enough confusion to make our own judgement call here.

The simplest behavior would be that if a host has confirmation enabled, it will send confirmations to any peer, regardless.

That would make the field in status-options (`ConfirmationsEnabled`) redundant, or at best just used for informative reasons (this node will send you confirmations), so I would remove it. 

I would not use it to indicate that the peer wants to receive information as it would overload the function of it, rather if we want that behavior I'd create a different field `WouldYouMindSendingMeConfirmationsPlease`)

// Digression starts
As a matter of fact, I would almost be tempted to couple `ConfirmationsEnabled` with `IsMailserver`, assuming that if it is a `Mailserver` it will send confirmations (I don't see a use case for a mailserver not sending confirmations, if the feature is not needed anymore, we can remove confirmations all together), but that's another story. 
At that point we could communicate the capabilities of a node in the handshake, i.e `IsMailserver`/`NodeInfo`, but I digress as not currently useful for any client, as mailservers are known in advance.
// Digression ends

TLDR:
I would remove `ConfirmationsEnabled` from the handshake and peer state, and remove the check in the code you posted above, leaving the behavior as simply:

If a host has `ConfirmationsEnabled` on, it will send confirmations to any peer.

Reply:
Thank you @cammellos, I'm pleased it wasn't just me being slow witted.

For the sake of this PR I'll make the amendment that `confirmations-enabled` means that the peer will send confirmations to the host.

If we want to discuss the removal of `confirmations-enabled` from the spec, I suppose we can do that in ever growing list of improvements for Waku V2. I can make an issue for these if you like.

I'll open an issue to track the check removal from the `status-go`, (otherwise I'll forget 😄).